test/Microsoft.AspNetCore.Mvc.IntegrationTests/ActionParametersIntegrationTest.cs

```diff
@@ -383,6 +383,120 @@ private class Person5
             Assert.Empty(modelState.Keys);
         }
 
+        [Fact]
+        public async Task ActionParameter_ModelPropertyTypeWithNoDefaultConstructor_NoOps()
+        {
+            // Arrange
+            var parameterType = typeof(Class1);
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "p",
+                ParameterType = parameterType
+            };
+            var testContext = ModelBindingTestHelper.GetTestContext(request =>
+            {
+                request.QueryString = QueryString.Create("Name", "James").Add("Property1.City", "Seattle");
+            });
+            var modelState = testContext.ModelState;
+
+            // Act
+            var result = await argumentBinder.BindModelAsync(parameter, testContext);
+
+            // Assert
+            Assert.True(result.IsModelSet);
+            Assert.True(modelState.IsValid);
+            var model = Assert.IsType<Class1>(result.Model);
+            Assert.Null(model.Property1);
+            var key = Assert.Single(modelState.Keys);
+            Assert.Equal("Name", key);
+            Assert.Equal("James", modelState[key].AttemptedValue);
+            Assert.Equal("James", modelState[key].RawValue);
```

Comment:
This is a bizarre pattern. If you copied this from another test, great if you could fix that too.
``` c#
var keyValuePair = Assert.Single(modelState);
Assert.Equal("Name", keyValuePair.Key);
Assert.Equal("James", keyValuePair.Value.AttemptedValue);
Assert.Equal("James", keyValuePair.Value.RawValue);
```

Much less important, good to specify `StringComparer.Ordinal` even in test assertions (when not ignoring case of course).